kmultiversx/src/kmultiversx/kasmer.py

```diff
@@ -151,7 +164,28 @@ def run_config_and_check_empty(
     return final_conf, sym_conf, subst
 
 
-def run_test(krun: KRun, sym_conf: KInner, init_subst: dict[str, KInner], endpoint: str, args: tuple[str, ...]) -> None:
+def run_pattern_and_check_exit_code(krun: KRun, conf: Pattern, pipe_stderr: bool = False) -> None:
```

Comment:
Maybe this works as well:

```suggestion
def run_pattern_and_check_exit_code(kprint: KPrint, conf: Pattern, pipe_stderr: bool = False) -> None:
```

Reply:
Also, consider hiding this function:

```suggestion
def _run_pattern_and_check_exit_code(krun: KRun, conf: Pattern, pipe_stderr: bool = False) -> None:
```

(And then can be moved after `run_test`,)

Reply:
Maybe everything except `run_concrete` should be private?